### ISSUE TITLE
Camera2Video glitching and green flickering in recorded video in Galaxy S6

### ISSUE THREAD

**Issue by buntupana** [android/camera-samples#348]
In Samsung Galaxy S6 the video result has a green flickering and glitching effect

[Gif of the issue](https://media.giphy.com/media/Ssj9BpyBZTea49NIa9/giphy.gif)

**Comment by buntupana** [comment]
@owahltinez any idea why this is happening?

**Comment by HasimD** [comment]
Yes. I am experiencing this problem too.

**Comment by buntupana** [comment]
@HasimD the only solution I found is change my implementation to use TextureView instead SurfaceView...just if it's useful for you. It's a bit of shame that Google doesn't provide a good examples and don't even attend the tickets created

**Comment by HasimD** [comment]
I agree with you,  i'm very surprised. The offical sample doesn't seem like professional. Also if I click to the capture button, the app crashes. I need to press it for a while. They said it is not an error and they will change it. Weird.

**Comment by nedols** [comment]
The app c

> 
> 
> I agree with you, i'm very surprised. The offical sample doesn't seem like professional. Also if I click to the capture button, the app crashes. I need to press it for a while. They said it is not an error and they will change it. Weird.

Looking at the code this is the correct behaviour. 
To start recording you need to tap&hold the capture button. When you release the capture button the application stop to recording video.

**Comment by amazingvoice** [comment]
I'm experiencing the same problem on my Galaxy s7 and Pixel 3. This is the only offical video recording demo using Camera2, which is embarrassing...

**Comment by amazingvoice** [comment]
@buntupana I just happened to find that an animation causes the white flash. Search for the variable _animationTask_ in _CameraFragment.kt_. Comment that out will solve the problem.

**Comment by buntupana** [comment]
@amazingvoice I'm not complaining about the white flickering, it's the green flickering and the glitching when the video is recorded what this ticket is about
